Add __next40pxDefaultSize for files in editor 3

AKSHAT2802 opened the pull request:
Part of - https://github.com/WordPress/gutenberg/issues/65018

## What?
Issue - https://github.com/WordPress/gutenberg/issues/65018, To use default to 40px for the button.


## Why?
To make the consistent button across Gutenberg, and we would have a lint rule added once fixed, all the button usage.


## How?
Add  __next40pxDefaultSize in button on the component.

## Testing Instructions
Screenshots added as comments in file changes

AKSHAT2802 committed:
Add __next40pxDefaultSize for files in editor 3

AKSHAT2802 commented on packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js:
```diff
@@ -428,8 +428,7 @@ export function HierarchicalTermSelector( { slug } ) {
 					<Button
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
  <tr>
    <td>
<img width="279" alt="term selector before" src="https://github.com/user-attachments/assets/f98c89ed-fdd0-449d-ae8f-d4eada176ea1">


</td>
    <td>
<img width="275" alt="term selector after" src="https://github.com/user-attachments/assets/83e223f6-6f1c-4562-aadb-b8c0ff31f11d">


</td>
  </tr>
</table>

Note:-  button variant set to link.

tyxla replied:
Looks good 👍 

@mirka, it felt off that we're forcing the size prop here. After all `variant="link"` buttons override the height to `auto` anyway? We're almost done with this migration, but if we were just beginning with it, maybe it would have made sense to exclude the `variant="link"` buttons from enforcing the size prop.

mirka replied:
Yeah, that makes sense. We'll make sure to exclude these when we start throwing deprecation warnings in console.

AKSHAT2802 commented on packages/editor/src/components/post-taxonomies/most-used-terms.js:
```diff
@@ -62,8 +62,7 @@ export default function MostUsedTerms( { onSelect, taxonomy } ) {
 						<Button
-							// TODO: Switch to `true` (40px size) if possible
-							__next40pxDefaultSize={ false }
+							__next40pxDefaultSize
```
<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
  <tr>
    <td>
<img width="271" alt="Most used term before" src="https://github.com/user-attachments/assets/583925dd-63de-461b-9659-83601af67d3c">

</td>
    <td>
<img width="271" alt="Most used term after" src="https://github.com/user-attachments/assets/7aaa43cb-b206-400e-a015-e13a9ec37166">


</td>
  </tr>
</table>
Note: Button variant set to link.

tyxla replied:
Ah, we have an awkward override here:

https://github.com/WordPress/gutenberg/blob/ecbde0e846ce3958f125302cfeda1cd34b5b89e3/packages/editor/src/components/post-taxonomies/style.scss#L38-L40

My preference: I'd just remove that override, and go with the `13px`. The default font size is 13px, so it wouldn't make much difference.

If we really wanted to make it smaller, we could use the `size="small"` prop, although I don't see a good reason to do it here.

AKSHAT2802 commented on packages/editor/src/components/post-url/index.js:
```diff
@@ -110,8 +110,7 @@ export default function PostURL( { onClose } ) {
 								<Button
-									// TODO: Switch to `true` (40px size) if possible
-									__next40pxDefaultSize={ false }
+									__next40pxDefaultSize
```
<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
  <tr>
    <td>
<img width="264" alt="post url before" src="https://github.com/user-attachments/assets/07527069-067c-4455-b681-b1be235a95ea">

</td>
    <td>
<img width="266" alt="post url after" src="https://github.com/user-attachments/assets/87ee9d9d-4762-4a79-8530-d91cf59bf457">


</td>
  </tr>
</table>

tyxla replied:
This was already defacto addressed as part of #64824 - `InputControl` would pass the `__next40pxDefaultSize` size prop along to the prefix and suffix component, and the underlying `Button` was obtaining it. 

It's likely you were using an old `trunk` when making the `Before` screenshot. In my testing, I see a 40px height button already, with and without this change. However, for consistency, it only makes sense we pass that prop here.

So, nothing to do here 👍

mirka replied:
Actually these screenshots are of a different UI. This file renders this UI here:

<img src="https://github.com/user-attachments/assets/5abbe914-02d3-4447-af4c-d5c404c735be" alt="Post URL popover" width="515">


☝️ Since this is in an InputControl prefix slot, we want this copy button to be `size="small"`, and wrap it in a `<InputControlSuffixWrapper variant="control">`.

Please see this [With Icon Or Control story](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--docs#with-icon-or-control) for an example.

tyxla commented on packages/editor/src/components/post-template/classic-theme.js:
```diff
@@ -177,8 +177,7 @@ function PostTemplateDropdownContent( { onClose } ) {
 					<Button
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
This one looks a bit awkward for a few reasons:

1. Inconsistent vertical alignment with the corresponding row label

The height of its label remains 32 px (20px line height + 6 px padding-top + 6 px padding-bottom), and it needs to be adapted to be 40px. This leads to poor vertical alignment of the label compared to the button:

![Screenshot 2024-09-11 at 15 53 33](https://github.com/user-attachments/assets/75a8f9f6-9de8-4a87-8b28-f4590a249b2c)

2. Inconsistent with the rest of the button instances we haven't yet migrated:

The height of the adjacent post summary row buttons is currently inconsistent:

![Screenshot 2024-09-11 at 15 59 00](https://github.com/user-attachments/assets/fc7da34f-e809-4c6c-9ecd-2ec44cbb7dfa)

The really awkward bit here is that ideally, both of these need to be addressed for all the `PostSummary` buttons simultaneously. Which means we might have missed an opportunity to group them together @mirka.

Perhaps we can go around it if we land them all together when they're ready?

ciampo replied:
> The height of its label remains 32 px (20px line height + 6 px padding-top + 6 px padding-bottom), and it needs to be adapted to be 40px. This leads to poor vertical alignment of the label compared to the button:

ideally we'd use a CSS layout that vertically centers the label without the need for hardcoded paddings/line heights. For example, `display: flex; align-items: center` on the parent container should do the trick

ciampo replied:
> Perhaps we can go around it if we land them all together when they're ready?

Sounds like a good idea, that could be a separate PR in itself

tyxla commented on packages/editor/src/components/save-publish-panels/index.js:
```diff
@@ -71,8 +71,7 @@ export default function SavePublishPanels( {
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
 					variant="secondary"
 					className="editor-layout__toggle-publish-panel-button"
```
I'd use the opportunity to remove this `className`. Doesn't seem to be used: https://wpdirectory.net/search/01J7GK5BMQPQ2B68RA3JBD2W57

tyxla commented on packages/editor/src/components/save-publish-panels/index.js:
```diff
@@ -86,8 +85,7 @@ export default function SavePublishPanels( {
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
 					variant="secondary"
 					className="editor-layout__toggle-entities-saved-states-panel-button"
```
Just as above, I'd remove this `className` as it's not in use: 

https://wpdirectory.net/search/01J7GK79JDYDBFFHA32Z5R9R7W

tyxla reviewed (commented):
Thanks for working on this one @AKSHAT2802 🙌 

Things mostly look good, but there are some opportunities for cleanup.

My biggest concern is the lack of consistency with the rest of the `PostSummary` buttons if we merge this one separately. Should we merge together with the rest? cc @mirka @ciampo

AKSHAT2802 committed:
Address feedback

AKSHAT2802 committed:
change font size to 13 px

tyxla commented on packages/editor/src/components/post-taxonomies/style.scss:
```diff
@@ -36,6 +36,6 @@
 
 	.components-button {
-		font-size: 12px;
+		font-size: 13px;
```
We should be able to remove this now, since 13px is the default font size.

AKSHAT2802 committed:
Revert classic-theme.js changes as it will be a seperate PR and revert 13px size in style as it is default

AKSHAT2802 commented:
Hii @mirka @tyxla 
I have removed redundant style for `font-size: 13px;` and reverted change for classic-theme.js file as it can be treated separately according to [this](https://github.com/WordPress/gutenberg/pull/65139#discussion_r1754442037).

mirka commented:
@AKSHAT2802 Could we also address this one please? https://github.com/WordPress/gutenberg/pull/65139#discussion_r1758655693

AKSHAT2802 committed:
Wrap copy button in  InputControlSuffixWrapper

mirka reviewed (approved):
Perfect, thanks!
